src/components/structures/RoomDirectory.js

```diff
@@ -56,7 +56,7 @@ export default createReactClass({
         };
     },
 
-    componentWillMount: function() {
+    componentDidMount: function() {
```

Comment:
from a quick skim this should be fine but all these member vars being set/initialized here could race.